dali/base/dafdesc.cpp

```diff
@@ -1351,6 +1355,7 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
             if (!tree)
                 tree = createPTree("Attr");
             attr.setown(tree);
+            history.setown(createPTree("History"));
```

Comment:
This is still a bit odd.
This constructor is supposed to be for when attr only has been passed (or null).

If history is required here, it would be clearer if you both/all initializers in a separate initializer method (that can be expanded if anything else were added in future and called here, so above code would say:
```
    if (tree)
        attr.setown(tree);
    ensureRequiredStructuresExist();
```

Where ensureRequiredStructuresExist would look like:
```
void ensureRequiredStructuresExist()
{
    if (!attr) attr.setown(createPTree("Attr"));
    if (!history) history.setown(createPTree("History"));
}
```

And perhaps ensureRequiredStructuresExist() could be reused elsewhere.